org.eclipse.wildwebdeveloper/src/org/eclipse/wildwebdeveloper/json/JSonLanguageServer.java

```diff
@@ -78,11 +82,50 @@ public void handleMessage(Message message, LanguageServer languageServer, URI ro
 		IPreferenceStore preferenceStore = Activator.getDefault().getPreferenceStore();
 		String schemaString = preferenceStore.getString(JSONPreferenceInitializer.JSON_SCHEMA_PREFERENCE);
 
-		Map<String, String> conversion = new Gson().fromJson(schemaString, new TypeToken<HashMap<String, String>>(){}.getType());
+		Map<String, String> conversion = new Gson().fromJson(schemaString, new TypeToken<HashMap<String, String>>() {
+		}.getType());
 
-		Map<String, List<String>> associations = new HashMap<>();
-		conversion.forEach((key, value) -> associations.put(key, Arrays.asList(value)));
+		Map<String, List<String>> schemaAssociations = new HashMap<>();
 
-		return associations;
+		IExtensionPoint point = Platform.getExtensionRegistry()
```

Comment:
Instead of looking directly in the plugin.xml, it would work better to use the content-type manager (`Platform.getContentTypeManager()`) to find out which content-types have json as basetype and read their `getFileSpec()` to infer the names and patterns.
The content-type registry can then be customized by other plugins or by end-users to add patterns and so so, which would be automatically passed to the Language Server.